src/H5Tcommit.c

```diff
@@ -337,6 +337,10 @@ H5Tcommit_anon(hid_t loc_id, hid_t type_id, hid_t tcpl_id, hid_t tapl_id)
     FUNC_ENTER_API(FAIL)
     H5TRACE4("e", "iiii", loc_id, type_id, tcpl_id, tapl_id);
 
+    /* Sanity check */
+    if (loc_id < 0 || type_id < 0 || tcpl_id < 0 || tapl_id < 0)
```

Comment:
Same as above. `loc_id`, `type_id`, `tcpl_id` and `tapl_id` are all checked immediately below.

Reply:
Fixed.